src/SocialAuthUserManager.php

```diff
@@ -490,7 +507,7 @@ public function loginUser(User $drupal_user) {
   protected function nullifySessionKeys() {
     if (!empty($this->sessionKeys)) {
       array_walk($this->sessionKeys, function ($session_key) {
-        $this->session->set($session_key, NULL);
+        $this->session->set($this->dataHandler->getSessionPrefix() . $session_key, NULL);
```

Comment:
You have a data handler object, you don't need a Symfony\Component\HttpFoundation\Session\Session object anymore. Instead, use the data handler to set, get, whatever data in session.

That being said, remove the $session service from this class